### ISSUE TITLE
Bug with citizens

### ISSUE THREAD

**Issue by Kryniowesegryderiusz** [tastybento/askyblock#557]
```[18:13:14] [Server thread/ERROR]: Could not pass event IslandEnterEvent to ASkyBlock v3.0.6
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot.jar:git-Spigot-de459a2-51263e9]
	at com.wasteofplastic.askyblock.listeners.PlayerEvents.onPlayerTeleport(PlayerEvents.java:760) [ASkyBlock.jar:?]
	at sun.reflect.GeneratedMethodAccessor108.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.craftbukkit.v1_10_R1.entity.CraftPlayer.teleport(CraftPlayer.java:459) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.craftbukkit.v1_10_R1.entity.CraftEntity.teleport(CraftEntity.java:226) [spigot.jar:git-Spigot-de459a2-51263e9]
	at net.citizensnpcs.npc.CitizensNPC.spawn(CitizensNPC.java:209) [citizens-2.0.21-SNAPSHOT.jar:?]
	at net.citizensnpcs.EventListen.spawn(EventListen.java:572) [citizens-2.0.21-SNAPSHOT.jar:?]
	at net.citizensnpcs.EventListen.respawnAllFromCoord(EventListen.java:549) [citizens-2.0.21-SNAPSHOT.jar:?]
	at net.citizensnpcs.EventListen.onChunkLoad(EventListen.java:139) [citizens-2.0.21-SNAPSHOT.jar:?]
	at sun.reflect.GeneratedMethodAccessor63.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot.jar:git-Spigot-de459a2-51263e9]
	at net.minecraft.server.v1_10_R1.Chunk.loadNearby(Chunk.java:910) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.craftbukkit.v1_10_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:53) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.craftbukkit.v1_10_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:1) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.craftbukkit.v1_10_R1.util.AsynchronousExecutor$Task.finish(AsynchronousExecutor.java:188) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.craftbukkit.v1_10_R1.util.AsynchronousExecutor.finishActive(AsynchronousExecutor.java:347) [spigot.jar:git-Spigot-de459a2-51263e9]
	at org.bukkit.craftbukkit.v1_10_R1.chunkio.ChunkIOExecutor.tick(ChunkIOExecutor.java:34) [spigot.jar:git-Spigot-de459a2-51263e9]
	at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:747) [spigot.jar:git-Spigot-de459a2-51263e9]
	at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot.jar:git-Spigot-de459a2-51263e9]
	at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot.jar:git-Spigot-de459a2-51263e9]
	at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot.jar:git-Spigot-de459a2-51263e9]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException
	at com.wasteofplastic.askyblock.GridManager.locationIsOnIsland(GridManager.java:1320) ~[?:?]
	at com.wasteofplastic.askyblock.listeners.PlayerEvents.processPerms(PlayerEvents.java:177) ~[?:?]
	at com.wasteofplastic.askyblock.listeners.PlayerEvents.onPlayerEnterOnIsland(PlayerEvents.java:166) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor554.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot.jar:git-Spigot-de459a2-51263e9]
	... 35 more
`

**Comment by tastybento** [comment]
You need to post more of the log to make it clear that this is an ASkyBlock issue - right now there's no indication that this is anything but a bug with Citizens and nothing to do with ASkyBlock.

**Comment by Kryniowesegryderiusz** [comment]
When I pasted it I saw that I haven't got half of error, so i enternd logs. I didnt expected that you would check it in 10 minutes :D

**Comment by tastybento** [comment]
Okay, it looks like the issue is that citizens don't have UUID's. Hmm. Can you give me a step by step to cause the issue? I'm not too familiar with Citizens. Did you just make one and put it on an island?

**Comment by tastybento** [comment]
Don't worry - I managed to get the error to occur. Let me work on it.

**Comment by Kryniowesegryderiusz** [comment]
Okey :D

**Comment by tastybento** [comment]
Fixed in 3.0.6.1.
